src/Generators/SubProfileGenerator.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Drupal\lightning\Generators;
+
+use Drupal\lightning\ComponentDiscovery;
+use DrupalCodeGenerator\Command\BaseGenerator;
+use DrupalCodeGenerator\Utils;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ChoiceQuestion;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+use Symfony\Component\Console\Question\Question;
+
+class SubProfileGenerator extends BaseGenerator
+{
+  /**
+   * The Lightning component discovery helper.
+   *
+   * @var \Drupal\lightning\ComponentDiscovery
+   */
+  protected $componentDiscovery;
```

Comment:
I don't mind continuing to use ComponentDiscovery for now (i.e., don't worry about fixing this right now), but we should eventually remove it. It was written aeons ago, when we thought we might have modules that don't start with `lightning_`. That never happened, though, so we could easily discover all Lightning modules by just filtering all modules from the `extension.list.module` service, looking for ones that start with `lightning_`.